serve_mux_test.go

```diff
@@ -70,3 +75,46 @@ func BenchmarkMuxMatch(b *testing.B) {
 	b.Run("lowercase", bench("_dns._udp.example.com."))
 	b.Run("uppercase", bench("_DNS._UDP.EXAMPLE.COM."))
 }
+
+func BenchmarkMuxMatchConcurrent(b *testing.B) {
+	domains := []string{
+		"example1.com",
+		"example2.com",
+		"example3.com",
+		"example4.com",
+		"example5.com",
+		"example6.com",
+		"example7.com",
+		"example8.com",
+		"example9.com",
+		"example10.com",
+	}
+
+	queries := []string{}
+	mux := NewServeMux()
+	for _, d := range domains {
+		mux.Handle(fmt.Sprintf("_udp.%s.", d), HandlerFunc(HelloServer))
+		queries = append(queries, fmt.Sprintf("_dns._udp.%s.", d))
+	}
+
+	threads := runtime.NumCPU()
+	perThread := b.N / threads
+	wg := sync.WaitGroup{}
+	wg.Add(threads)
+
+	b.ResetTimer()
+
+	for i := 0; i < threads; i++ {
+		go func() {
+			defer wg.Done()
+			for n := 0; n < perThread; n++ {
+				handler := mux.match(queries[n%len(queries)], TypeSRV)
+				if handler == nil {
+					b.Fatal("couldn't find match")
```

Comment:
You can’t call `Fatal` in another goroutine.

Reply:
Woops, yep, you're right